lib/ansible/modules/packaging/os/yum.py

```diff
@@ -583,6 +583,9 @@ def local_nvra(module, path):
     fd = os.open(path, os.O_RDONLY)
     try:
         header = ts.hdrFromFdno(fd)
+    except:
+        if module.params.get('skip_broken'):
+            return None
```

Comment:
The exception needs to be re-raised if `skip_broken` is not set, otherwise it's lost.